src/main/kotlin/org/rust/lang/core/resolve/ref/RsLitExprReferenceContributor.kt

```diff
@@ -62,3 +73,70 @@ private class RsLiteralFileReferenceSet(
         }
     }
 }
+
+private val pathValueLiteral: PsiElementPattern.Capture<RsLitExpr> = psiElement<RsLitExpr>()
+    .with("onPathLiteral") { expr -> expr.kind is RsLiteralKind.String && isPathLitExpr(expr) }
+
+private fun isPathLitExpr(expr: RsLitExpr): Boolean {
+    val (function, arguments) = getFunctionAndArguments(expr) ?: return false
+    val parentImpl = function.owner as? RsAbstractableOwner.Impl
```

Comment:
We do not want to return here, otherwise it would not work for free functions. I don't understand the CI error though, I thought that if I use `as?`, it will return null if the cast cannot proceed. There is no unsafe cast, so I wonder why it crashes.

Reply:
Sorry, my mistake. You are right

Reply:
> There is no unsafe cast, so I wonder why it crashes.

But I can't see failed tests on CI

Reply:
Nevermind, the previous commit failed, I fixed it via your suggestion before the CI run finished and then I received an e-mail with the error :-)